### PR TITLE
added pipepine, implement event logging system, refactored main.py, added requirements.txt

### PR DESCRIPTION
**Pull request by NazarKarpiuk-stud**
(My apologies. I closed previous pull request due to mistake in last commit.)

Created pipeline-tasks.yaml in workflow, where I placed scripts for unit-tests auto-running and extracting the artifacts(part of log file) from project. Also, created feature that allows user download those artifacts (in form of archive) and review the results on their own.

### CODE REVIEW

**Comment by NazarKarpiuk-stud** [comment]
Saw the remaining problem in .github/workflows/pipeline-tasks.yaml. Will fix it in next update.

File: .github/workflows/pipeline-tasks.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: pipeline-tasks
```
Comment (andrii-s-shpitser-lpnu-ua) [inline .github/workflows/pipeline-tasks.yaml:1]: what task is it?

Comment (NazarKarpiuk-stud) [reply .github/workflows/pipeline-tasks.yaml:1]: Fourth task. Creating pipelines.

Comment (andrii-s-shpitser-lpnu-ua) [reply .github/workflows/pipeline-tasks.yaml:1]: this file already done by other student and will not be estimated. Please close PR and start next task.

Comment (NazarKarpiuk) [reply .github/workflows/pipeline-tasks.yaml:1]: This is completely my file there is no way someone could make it. I also added this file and tested it running on my other repository about ten hours ago. Here is the proofs: https://github.com/NazarKarpiuk/test_repos/tree/main/.github/workflows

**Pull request closed without merging**